libcore/tests/UtilTests/UtilTests.vala

```diff
@@ -346,6 +346,64 @@ void add_file_utils_tests () {
         assert (!result.contains (Files.FileUtils.COPY_TAG));
         assert (result.contains ("2"));
     });
+
+    Test.add_func ("/FileUtils/disambiguate_path/conflicting", () => {
+        var result = Files.FileUtils.disambiguate_uri (
+            "/A/B/folder",
+            "/A/C/folder"
+        );
+        assert (result == "B/folder");
+    });
+
+    Test.add_func ("/FileUtils/disambiguate_path/different_length", () => {
+        var result = Files.FileUtils.disambiguate_uri (
+            "/A/B/folder",
+            "X/A/B/folder"
+        );
+        assert (result == "A/B/folder");
```

Comment:
Should this actually be "/A/B/folder" for the expected result since it implies an absolute path rather than relative?

Reply:
Yes - you are right!  This function is supposed to return enough  to distinguish the two paths.  I'll fix.